src/plugins/expressions/common/execution/types.ts

```diff
@@ -59,6 +62,13 @@ export interface ExecutionContext<
    */
   getSearchSessionId: () => string | undefined;
 
+  /**
+   * Makes a `KibanaRequest` object available to expression functions. Useful for
+   * functions which are running on the server and need to perform operations that
+   * are scoped to a specific user.
+   */
+  kibanaRequest?: KibanaRequest;
```

Comment:
whats the latest status of this KibanaRequest, at some point we were discussing that it would be nice to have something that doesn't contain full request but just the scoped user information ?

Reply:
Given the current design of our services I don't think we can avoid this -- as long as there is at least one service requiring a full `KibanaRequest`, we'd need to provide the whole thing to expressions.

Currently the low level search takes a full kibana request on the server since it gets passed along to all search strategies. This also means that high level search (search source) needs a scoped request too. So `esaggs` will need a full kibana request either way.

The services that don't currently use a full request but just take in scoped services are aggs, index patterns, field formats. [Matt and I were recently discussing](https://github.com/elastic/kibana/pull/81218#discussion_r533761497) how this is a consistency-vs-explicitness tradeoff, and I think both sides have pros & cons.